sheets/google_sheets/download_file.py

```diff
@@ -0,0 +1,268 @@
+    """
+    Use the file name provided in the source_file_name variable. Should be run only
+    if a destination_file_name is not provided.
+    """
+    destination_file_name = os.path.basename(source_full_path)
+    return destination_file_name
+
+
+def clean_folder_name(folder_name):
+    """
+    Cleans folders name by removing duplicate '/' as well as leading and trailing '/' characters.
+    """
+    folder_name = folder_name.strip('/')
+    if folder_name != '':
+        folder_name = os.path.normpath(folder_name)
+    return folder_name
+
+
+def combine_folder_and_file_name(folder_name, file_name):
+    """
+    Combine together the provided folder_name and file_name into one path variable.
+    """
+    combined_name = os.path.normpath(
+        f'{folder_name}{"/" if folder_name else ""}{file_name}')
+    combined_name = os.path.normpath(combined_name)
+
+    return combined_name
+
+
+def determine_destination_name(destination_folder_name, destination_file_name, ):
+    """
+    Determine the final destination name of the file being downloaded.
+    """
+    destination_name = combine_folder_and_file_name(
+        destination_folder_name, destination_file_name)
+    return destination_name
+
+
+def find_google_cloud_storage_file_names(bucket, prefix=''):
+    """
+    Fetched all the files in the bucket which are returned in a list as 
+    Google Blob objects
+    """
+    return list(bucket.list_blobs(prefix=prefix))
+
+
+def find_matching_files(file_blobs, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for blob in file_blobs:
+        if re.search(file_name_re, blob.name):
+            matching_file_names.append(blob)
+
+    return matching_file_names
+
+
+def download_google_sheet_file(
+        service,
+        spreadsheet_id,
+        tab_name,
+        cell_range,
+        destination_file_name=None):
+    """
+    Download th contents of a spreadsheet from Google Sheets to local storage in
+    the current working directory.
+    """
+    local_path = os.path.normpath(f'{os.getcwd()}/{destination_file_name}')
+    try:
+        if tab_name:
+            if check_workbook_exists(service=service,
+                                    spreadsheet_id=spreadsheet_id,
+                                    tab_name=tab_name):
+                cell_range = f'{tab_name}!{cell_range}'
+        sheet = service.spreadsheets().values().get(
+                                    spreadsheetId=spreadsheet_id,
+                                    range=cell_range).execute()
+
+        if not sheet.get('values'):
+            print(f'No values for {spreadsheet_id}.. Not downloading')
+            return
+
+        values = sheet['values']
+        with open(local_path, '+w') as f:
+            writer = csv.writer(f)
+            writer.writerows(values)
+        print(f'Successfully {spreadsheet_id} downloaded to {local_path}')
```

Comment:
Can we change this to 'Successfully downloaded contents of Spreadsheet {spreadsheet_name} - Sheet {sheet_name} to {local_path}'?